src/integration_tests/rpc_multiple_remote_addrs/main.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016 Alexander Gallego. All rights reserved.
+//
+// std
+#include <chrono>
+#include <iostream>
+// linux
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+// seastar
+#include <core/app-template.hh>
+#include <core/distributed.hh>
+#include <core/sleep.hh>
+#include <net/api.hh>
+// smf
+#include "integration_tests/non_root_port.h"
+#include "smf/log.h"
+#include "smf/random.h"
+#include "smf/rpc_generated.h"
+#include "smf/rpc_handle_router.h"
+#include "smf/rpc_recv_context.h"
+#include "smf/rpc_server.h"
+// templates
+#include "integration_tests/demo_service.smf.fb.h"
+
+// For a network connected host we should always have at least an lo and eth0 interface 
+// (or whatever the connected interface is called) ... so capping the maximum test clients
+// at 2 will cover this. Going larger doesn't improve the accuracy of this test.
+constexpr static const size_t kMaxClients = 2;
+
+struct client_info {
+    explicit client_info(seastar::ipv4_addr ip): client(smf_gen::demo::SmfStorageClient::make_shared(ip)), ip(ip.ip) {};
+    client_info(){};
```

Comment:
you should define a move constructor. and remove the copy constructor - otherwise ip_str *could* leak by pointing the char to a memory not managed by this class.